addons/mail/static/src/new/discuss/discuss.js

```diff
@@ -165,12 +165,16 @@ export class Discuss extends Component {
 
     async renameGuest({ value: name }) {
         const newName = name.trim();
-        if (this.store.guest?.name !== newName) {
+        if (this.store.guest && this.store.guest.name !== newName) {
             await this.personaService.updateGuestName(this.store.self, newName);
         }
     }
 
-    get guestAvatarUrl() {
-        return `/mail/channel/${this.thread.id}/guest/${this.store.guest.id}/avatar_128?unique=${this.store.guest.name}`;
+    get avatarUrl() {
+        if (this.store.guest) {
+            return `/mail/channel/${this.thread.id}/guest/${this.store.guest.id}/avatar_128?unique=${this.store.guest.name}`;
+        } else {
+            return this.store.user.avatarUrl;
+        }
```

Comment:
This becomes almost the same as `avatarUrl` in `persona_model`.

It goes in `mail/channel` route for guest just for access rights.
Would be nice to clean code of `avatarUrl`, as there are too many different uses that should be reconciled

Reply:
Follow up PR :)